Flaky CI failure in WordExtractor close test

### Self Checks

- [x] I have read the [Contributing Guide](https://github.com/langgenius/dify/blob/main/CONTRIBUTING.md) and [Language Policy](https://github.com/langgenius/dify/issues/1542).
- [x] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [x] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [x] I confirm that I am using English to submit this report, otherwise it will be closed.
- [x] 【中文用户 & Non English User】请使用英语提交，否则会被关闭 ：）
- [x] Please do not modify this template :) and fill in all the required fields.

### Dify version

main

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce


  The API unit test sometimes fails in CI:

  test_close_handles_async_close_mock
  RuntimeWarning: coroutine 'AsyncMockMixin._execute_mock_call' was never awaited

  This is hard to reproduce locally because it depends on test order and garbage collection timing.

  The test used gc.collect() inside warnings.catch_warnings(). This can catch warnings from async mock coroutines leaked by earlier tests in the
  same pytest worker, not only warnings from this test.

### ✔️ Expected Behavior

The test should only check WordExtractor.close() behavior.

### ❌ Actual Behavior
```
FAILED api/tests/unit_tests/core/rag/extractor/test_word_extractor.py::test_close_handles_async_close_mock - assert not [<warnings.WarningMessage object at 0x7f2eb5509940>]
```
github action error below
https://github.com/langgenius/dify/actions/runs/25037138795/job/73331450574
https://github.com/langgenius/dify/actions/runs/25055872851/job/73396113155

The test can fail because it captures unrelated warnings from previous tests.

--- a/api/tests/unit_tests/core/rag/extractor/test_word_extractor.py
+++ b/api/tests/unit_tests/core/rag/extractor/test_word_extractor.py
@@ -1,14 +1,12 @@
 """Primarily used for testing merged cell scenarios"""
 
-import gc
 import io
 import os
 import tempfile
-import warnings
 from collections import UserDict
 from pathlib import Path
 from types import SimpleNamespace
-from unittest.mock import AsyncMock, MagicMock
+from unittest.mock import MagicMock
 
 import pytest
 from docx import Document
@@ -377,23 +375,21 @@ def test_close_is_idempotent():
     extractor.temp_file.close.assert_called_once()
 
 
-def test_close_handles_async_close_mock():
+async def _async_close() -> None:
+    return None
+
+
+def test_close_closes_awaitable_close_result():
     extractor = object.__new__(WordExtractor)
     extractor._closed = False
     extractor.temp_file = MagicMock()
-    extractor.temp_file.close = AsyncMock()
+    close_result = _async_close()
+    extractor.temp_file.close = MagicMock(return_value=close_result)
 
-    with warnings.catch_warnings(record=True) as caught:
-        warnings.simplefilter("always")
-        extractor.close()
-        gc.collect()
+    extractor.close()
 
+    assert close_result.cr_frame is None
     extractor.temp_file.close.assert_called_once()
-    assert not [
-        warning
-        for warning in caught
-        if issubclass(warning.category, RuntimeWarning) and "AsyncMockMixin._execute_mock_call" in str(warning.message)
-    ]
 
 
 def test_extract_images_handles_invalid_external_cases(monkeypatch):

